From 8037fd13b7e2f44ce965c9dabaede236168e1f31 Mon Sep 17 00:00:00 2001
From: gnome-mpv <gnome-mpv@nowhere>
Date: Sat, 19 Aug 2023 23:04:20 +0700
Subject: [PATCH] player: Fix crash when mpv-config-file isn't set to a URI

Fixes #880.
---
 src/celluloid-player.c | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/src/celluloid-player.c b/src/celluloid-player.c
index 06804599..9eb78cf5 100644
--- a/src/celluloid-player.c
+++ b/src/celluloid-player.c
@@ -788,16 +788,26 @@ load_config_file(CelluloidMpv *mpv)
 
 	if(g_settings_get_boolean(settings, "mpv-config-enable"))
 	{
-		gchar *mpv_conf =	g_settings_get_string
-					(settings, "mpv-config-file");
+		gchar *mpv_conf =
+			g_settings_get_string(settings, "mpv-config-file");
 
 		GFile *file = g_file_new_for_uri(mpv_conf);
 		gchar *path = g_file_get_path(file);
 
-		g_info("Loading config file: %s", path);
-		celluloid_mpv_load_config_file(mpv, path);
+		g_info("Loading mpv config file: %s", mpv_conf);
+
+		if(path)
+		{
+			g_debug("mpv config file path: %s", path);
+			celluloid_mpv_load_config_file(mpv, path);
+
+			g_free(path);
+		}
+		else
+		{
+			g_warning("Failed to load mpv config file");
+		}
 
-		g_free(path);
 		g_object_unref(file);
 		g_free(mpv_conf);
 	}
